src/utils/retryFetch.test.ts

```diff
@@ -0,0 +1,26 @@
+import { retryFetch } from "./retryFetch";
+
+describe("[retryFetch]", () => {
+  describe("when there are retries configured", () => {
+    it("should exaust retries", async () => {
+      const maxRetries = 3;
+      const loader = jest.fn(() => Promise.reject("error"));
+
+      try {
+        retryFetch({ loader, options: { delay: 100, maxRetries } });
```

Comment:
There is [a way to check a promise rejects](https://jestjs.io/docs/tutorial-async#rejects) in jest, might be good to use here?